SDWebImage/SDImageCache.m

```diff
@@ -378,6 +379,10 @@ - (nullable NSOperation *)queryCacheOperationForKey:(nullable NSString *)key opt
     
     // First check the in-memory cache...
     UIImage *image = [self imageFromMemoryCacheForKey:key];
+    if ((options & SDImageCacheDecodeFirstFrameOnly) && image.sd_isAnimated) {
+        image = image.sd_firstFrame;
+    }
```

Comment:
`animatedImageFrameAtIndex:` method can get the first frame.

Reply:
got it, Thanks!!

Reply:
Should we consider the `SDAnimatedImage` case here ?

If so, maybe this is better...This also handle the `SDAnimatedImage`, `UIAnimatedImage` and `UIImage(static)`...


```
    if ((options & SDImageCacheDecodeFirstFrameOnly) && image.sd_isAnimated) {
#if SD_MAC
        image = [[NSImage alloc] initWithCGImage:image.CGImage scale:image.scale orientation:kCGImagePropertyOrientationUp];
#else
        image = [[UIImage alloc] initWithCGImage:image.CGImage scale:image.scale orientation:image.orientation];
#endif
```

Reply:
It makes sense, Updated.
and I no any updates.